Step 4: add support for unique with dim=None

zasdfgbnm opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #18661 Step 7: remove _unique
* #18655 Step 6: Rename _unique2 to unique and add int? dim
* #18654 Step 5: remove _unique_dim in favor of unique_dim
* **#18651 Step 4: add support for unique with dim=None**

See step 1 for the description of the full stack

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [D15000463](https://our.internmc.facebook.com/intern/diff/D15000463)

zasdfgbnm committed:
Step 4: add support for unique with dim=None

See step 1 for the description of the full stack

zasdfgbnm committed:
Update on "Step 4: add support for unique with dim=None"

Step 4: add support for unique with dim=None

See step 1 for the description of the full stack

gh-metadata: pytorch pytorch 18651 gh/zasdfgbnm/4/head

zasdfgbnm committed:
Update on "Step 4: add support for unique with dim=None"

Step 4: add support for unique with dim=None

See step 1 for the description of the full stack

gh-metadata: pytorch pytorch 18651 gh/zasdfgbnm/4/head

zasdfgbnm committed:
Update on "Step 4: add support for unique with dim=None"

Step 4: add support for unique with dim=None

See step 1 for the description of the full stack

gh-metadata: pytorch pytorch 18651 gh/zasdfgbnm/4/head

zasdfgbnm committed:
Update on "Step 4: add support for unique with dim=None"

Step 4: add support for unique with dim=None

See step 1 for the description of the full stack

gh-metadata: pytorch pytorch 18651 gh/zasdfgbnm/4/head

zasdfgbnm committed:
Update on "Step 4: add support for unique with dim=None"

Step 4: add support for unique with dim=None

See step 1 for the description of the full stack

gh-metadata: pytorch pytorch 18651 gh/zasdfgbnm/4/head

zasdfgbnm committed:
Update on "Step 4: add support for unique with dim=None"

Step 4: add support for unique with dim=None

See step 1 for the description of the full stack

gh-metadata: pytorch pytorch 18651 gh/zasdfgbnm/4/head

zasdfgbnm committed:
Update on "Step 4: add support for unique with dim=None"

Step 4: add support for unique with dim=None

See step 1 for the description of the full stack

gh-metadata: pytorch pytorch 18651 gh/zasdfgbnm/4/head

zasdfgbnm commented:
@VitalyFedyunin step 4 and 5 should also be ready

VitalyFedyunin commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -1929,7 +1929,8 @@
 
-- func: _unique2_temporary_will_remove_soon(Tensor self, bool sorted=True, bool return_inverse=False, bool return_counts=False) -> (Tensor, Tensor, Tensor)
+- func: _unique2(Tensor self, bool sorted=True, bool return_inverse=False, bool return_counts=False) -> (Tensor, Tensor, Tensor)
+  matches_jit_signature: True
```
This is now default value, please remove

VitalyFedyunin commented:
Please rebase upon master it will help to import changes.

zasdfgbnm committed:
Update on "Step 4: add support for unique with dim=None"

Step 4: add support for unique with dim=None

See step 1 for the description of the full stack

gh-metadata: pytorch pytorch 18651 gh/zasdfgbnm/4/head

zasdfgbnm commented:
@VitalyFedyunin All fixed

zasdfgbnm committed:
Update on "Step 4: add support for unique with dim=None"

Step 4: add support for unique with dim=None

See step 1 for the description of the full stack

gh-metadata: pytorch pytorch 18651 gh/zasdfgbnm/4/head

VitalyFedyunin commented:
Thanks! There are couple tests I'm still waiting to finish and I will land afterwards.
